### PR TITLE
Adds templates for doc building workflows

### PR DESCRIPTION
**Pull request by sgugger**
This PR adds templates for the jobs that:
- build the doc at each commit on master
- build the doc for each PR
- clean the doc after each PR

The last two have been tested on Accelerate. The first one still needs testing.

### CODE REVIEW

File: .github/workflows/build_pr_documentation.yml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        type: string
+      pr_number:
+        required: true
+        type: string
+      package:
+        required: true
+        type: string
+      path_to_doc:
+        type: string
+
```
Comment (mishig25) [inline .github/workflows/build_pr_documentation.yml:17]: in [transformers pr_doc](https://github.com/huggingface/transformers/blob/master/.github/workflows/build_dev_documentation.yml#L6-L8), there are lines about concurrency

```py
concurrency:
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
  cancel-in-progress: true
```

do we need them here as well?

File: .github/workflows/build_main_documentation.yml [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+        type: string
+      path_to_doc:
+        type: string
+      notebook_folder:
+        type: string
+    secrets:
+      token:
+        required: true
+
+jobs:
```
Comment (mishig25) [inline .github/workflows/build_main_documentation.yml:20]: in transformers [build doc workkflow](https://github.com/huggingface/transformers/blob/master/.github/workflows/build_documentation.yml#L45-L53), there is a `job` about `cache`:
```yml
      - name: Loading cache.
        uses: actions/cache@v2
        id: cache
        with:
          path: ~/.cache/pip
          key: v1-test_build_doc
          restore-keys: |
            v1-test_build_doc-${{ hashFiles('setup.py') }}
            v1-test_build_doc
```

do we need them here as well? or add them as a `workflow.inputs`

Comment (sgugger) [reply .github/workflows/build_main_documentation.yml:20]: No, this is using the docker image now.

File: .github/workflows/build_pr_documentation.yml (hunk 1 above)
Comment (sgugger) [reply .github/workflows/build_pr_documentation.yml:17]: This should be on the workflows that apply this template (if the library wants it), see for instance in [this PR](https://github.com/huggingface/doc-builder/pull/138) (GitHub doesn't let me link to a specific line but it's line 6 in the build_pr_documentation file)

**Review by mishig25 — APPROVED** [review]

File: .github/workflows/build_main_documentation.yml [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      package:
+        required: true
+        type: string
+      path_to_doc:
+        type: string
+      notebook_folder:
+        type: string
+    secrets:
+      token:
+        required: true
```
Comment (LysandreJik) [inline .github/workflows/build_main_documentation.yml:18]: Very clean

**Review by LysandreJik — APPROVED** [review]
Looks great, thanks for spending some time on this!

**Pull request merged**